### DISCUSSION TITLE
How can I correctly free a pointer in unmanaged memory from managed code?

### DISCUSSION THREAD

**Discussion by ankofl** [CGAL/cgal#8526] [Q&A] [answered]
Is this example correct?
```c
extern "C" {
    __declspec(dllexport) void extern_solid_release(void* ptr) {
        if (ptr) { // Проверяем, что указатель не нулевой
            Mesh* mesh = reinterpret_cast<Mesh*>(ptr);
            delete mesh;
            mesh = nullptr; // Обнуляем указатель после удаления
        }
    }
}
```

```cs
  [DllImport(CallerCGAL.pathDll, CallingConvention = CallingConvention.Cdecl)]
  private static extern void extern_solid_release(IntPtr ptr);
```

**Comment by lrineau** [comment]
Hi @ankofl. That is difficult to understand why that is related to CGAL. I see the word `CallerCGAL` in your code, but I do not know what it is.

Actually, I do not even know which language you are programming into. I doubt anybody here can help you. Maybe ask on StackOverflow?

**Reply by ankofl** [reply, marked as answer]
I apologize for the ambiguity

I write in C# NET.9 and I call C++ libraries compiled using header libraries via DllImport .h-files of CGAL. I am interested in the question: when I created a Mesh object (Polyhedron_3) in C++ code and sent a pointer to this object to C# code, how would it be correct to free memory from C# inside the unmanaged C++ code in which the Mesh object was created? 

I tried to do it via:
```cpp
delete mesh;
mesh = null ptr;
```

But I am not sure that this will be correct. 

Maybe Polyhedron_3 and Surface_Mesh objects have specialized methods to free up the memory they occupy? 

Thanks!

**Reply by lrineau** [reply]
When you call delete that way, the destructor method of the pointed object is called, and the the memory freed. That should be fine.

**Reply by ankofl** [reply]
Thank you very much!
